great_expectations/data_context/types/base.py

```diff
@@ -2373,13 +2369,10 @@ def __init__(  # noqa: C901, PLR0912, PLR0913, PLR0915
         self.expectations_store_name = expectations_store_name
         self.validations_store_name = validations_store_name
         self.evaluation_parameter_store_name = evaluation_parameter_store_name
-        if checkpoint_store_name is not None:
-            self.checkpoint_store_name = checkpoint_store_name
-        if profiler_store_name is not None:
-            self.profiler_store_name = profiler_store_name
+        self.checkpoint_store_name = checkpoint_store_name
+        self.profiler_store_name = profiler_store_name
         self.plugins_directory = plugins_directory
-        if validation_operators is not None:
-            self.validation_operators = validation_operators
```

Comment:
These feel bad. I shouldn't have the reason about whether or not an object has an attr. Shape should be consistent (even if values are not!)